core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -59,23 +59,34 @@ public GitChangeResolver(File projectDir, Scm scm) {
     public GitChangeResolver(File dir, String previous, String head) {
         this.previous = previous;
         this.head = head;
-        final FileRepositoryBuilder builder = new FileRepositoryBuilder();
-        try {
-            this.git = new Git(builder.readEnvironment().findGitDir(dir).build());
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Unable to find git repository for path " + dir.getAbsolutePath(), e);
+        final FileRepositoryBuilder fileRepositoryBuilder = getFileRepositoryBuilder(dir);
+        if (fileRepositoryBuilder.getGitDir() != null) {
+            try {
+                this.git = new Git(fileRepositoryBuilder.build());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            this.repoRoot = git.getRepository().getDirectory().getParentFile();
+        } else {
+            logger.warn(String.format("Unable to find a git repository for the path %s - new, changed, affected strategies won't work properly."
```

Comment:
I wouldn't mention the concrete names of the strategies - if we change a name or add a new one, we won't probably remember that should be changed in here...